src/Notification/ToastNotification.svelte

```diff
@@ -66,14 +69,17 @@
   }
 
   onMount(() => {
-    if (timeout) {
-      timeoutId = setTimeout(() => close(true), timeout);
-    }
-
     return () => {
       clearTimeout(timeoutId);
     };
   });
+
+  $: if (typeof window !== "undefined") {
+    /** Only start the timer of {@link open} has not been set to `false`. */
+    if (open && timeout) {
```

Comment:
```suggestion
    if (open && timeout && timeoutId === undefined) {
```

> While the toast is open, the timeout cannot be disabled by setting the property to 0 and it cannot be changed to something else without starting a parallel timeout (with the old one not being cleaned up).

@brunnerh Thanks for the review – regarding your latest feedback, WDYT about adding an additional check that a timer has not already been set?

Reply:
I would add this:
```diff
  $: if (typeof window !== "undefined") {
+   clearTimeout(timeoutId);

    /** Only start the timer of {@link open} has not been set to `false`. */
    if (open && timeout) {
```

If the timeout is changed to 0, it cancels but no new timeout is started.
If the timeout is changed to something else, it also cancels and starts a new one (unless the toast is already closed).

<details>
<summary>Test code I used for switching timeouts</summary>

```svelte
<script>
    import { ToastNotification } from 'carbon-components-svelte';

	let timeout = 3000;
	let events = [];
</script>

<button on:click={() => timeout = 1000}>1s</button>
<button on:click={() => timeout = 3000}>3s</button>
<button on:click={() => timeout = 0}>disable</button>

<ToastNotification
	title="Auto-Close"
	{timeout}
	on:close={e => (events = [...events, e.detail])}>
	<svelte:fragment slot="subtitle">
		Timeout is {timeout}ms
	</svelte:fragment>
</ToastNotification>

<div>
	{#each events as event, i}
		<pre>{JSON.stringify(event, null, 2)}</pre>
	{/each}
</div>
```
</details>

Reply:
Makes sense – thanks. Amended the commit with you as a co-author.